### PR TITLE
Switch to preact

### PR DESCRIPTION
**Pull request by kof**

### CODE REVIEW

**Comment by kof** [comment]
Whats still todo here?

**Comment by kof** [comment]
Lets make @_developit happy and switch to preact)))

**Comment by kof** [comment]
Lets make @developit happy and switch to preact)))

**Comment by typical000** [comment]
@kof let's switch to preact. But before resolve conflict and make final docs build.

**Comment by developit** [comment]
Keep me posted on this as you merge 👍

**Comment by typical000** [comment]
Updated dists, updated version.

But we can't remove "react" and "react-dom" from dependencies because we use them in server.js where aliases such `react <==> preact-compat` are not available.
But, anyway, our vendor bundle decreased in size from 163kb up to 130kb gzipped. :)

**Comment by developit** [comment]
@typical000 Nice size decrease!  How come the aliases aren't available in `server.js`?

I've seen people do this to enable them using [module-alias](https://www.npmjs.com/package/module-alias):

```js
import moduleAlias from 'module-alias';

moduleAlias.addAliases({
  react: 'preact-compat',
  'react-dom': 'preact-compat'
});

// as before
```

you can even pull them from your webpack config:

```js
require('module-alias').addAliases(require('../webpack.config.base').resolve.alias);

// as before
```

**Comment by typical000** [comment]
@developit Cool! Used `module-alias` as you suggested! Thank you
Now ssr build works fine! Removed completely `react` and `react-dom`

**Comment by kof** [comment]
Should I merge it?

**Comment by developit** [comment]
I literally just used this technique, so happy to have found it.  I think I stole it from next.js?

**Comment by typical000** [comment]
@kof I think that - yes. Fixed last lint issues just now. `build-html, build:prod, build:dev` works fine

File: src/components/HighlightedMarkdown/H.js [hunk 1]
```diff
@@ -16,12 +24,12 @@ export default class H extends PureComponent {
...
   }
 
-  onClick = () => {
+  onClick() {
     // We need to set the hash manually because <Link/> will not.
     // Also we can't use `onSetActive` callback because it doesn't work if
     // an element is on very bottom so that there is no scroll possible.
     setTimeout(() => {
-      location.hash = this.props.children[0].props.to
+      location.hash = this.to
```
Comment (kof) [inline src/components/HighlightedMarkdown/H.js]: where is `this.to` defined? this is very implicit, we can do better!

Comment (developit) [reply src/components/HighlightedMarkdown/H.js]: seems to be a property on the DOM element - personally I recommend the approach you guys were doing before, using the class-bound arrow methods and reaching into props.  VDOM scales a lot more nicely when the DOM is treated as write-only.

Comment (typical000) [reply src/components/HighlightedMarkdown/H.js]: @kof you right. It is very implict. But after switching to preact there is no "hidden" change of `this.props.children` - it remains the same as passed from parent component, e.g. - just a string without attached link.

**Comment by kof** [comment]
merged!

**Pull request merged**